File: FitToFit/FitToFit.Services/Database/Novosti.cs
using System;
using System.Collections.Generic;

namespace FitToFit.Services.Database;

public partial class Novosti
{
    public int NovostId { get; set; }

    public string Naslov { get; set; } = null!;

    public string? Sadrzaj { get; set; }

    public bool? IsLiked { get; set; }

    public int BrojLajkova { get; set; }

    public DateTime DatumObjave { get; set; }

    public int KorisnikId { get; set; }

    public int? VrstaTreningaId { get; set; }

    public bool? IsRead { get; set; }

    public virtual ICollection<KorisniciNovosti> KorisniciNovostis { get; set; } = new List<KorisniciNovosti>();

    public virtual Korisnici Korisnik { get; set; } = null!;

    public virtual VrsteTreninga? VrstaTreninga { get; set; }
}

File: FitToFit/UI/fittofit_admin/lib/models/novosti.dart
import 'package:json_annotation/json_annotation.dart';

part 'novosti.g.dart';

@JsonSerializable()
@JsonKey(name: 'isLiked')
class Novosti{
  int? novostId;
  String? naslov;
  String? sadrzaj;
  bool? isLiked;
  int? brojLajkova;
  DateTime? datumObjave;
  int korisnikId;
  int? vrstaTreningaId;

  Novosti(this.novostId, this.naslov, this.sadrzaj, this.isLiked, this.brojLajkova, this.datumObjave, this.korisnikId, this.vrstaTreningaId);

  factory Novosti.fromJson(Map<String, dynamic> json) => _$NovostiFromJson(json);

  Map<String, dynamic> toJson() => _$NovostiToJson(this);
}